packages/react-instantsearch-dom-maps/src/__tests__/Provider.js

```diff
@@ -444,11 +432,99 @@ describe('Provider', () => {
 
       const wrapper = shallow(<Provider {...props}>{x => x}</Provider>);
 
-      expect(wrapper.instance().getChildContext()).toEqual({
-        [STATE_CONTEXT]: expect.objectContaining({
-          refineWithInstance: wrapper.instance().refineWithInstance,
-        }),
+      expect(wrapper.props().value.refineWithInstance).toBe(
+        wrapper.instance().refineWithInstance
+      );
+    });
+
+    it('expect to pass new mapValue reference when isRefineOnMapMove changes', () => {
+      const props = {
+        ...defaultProps,
+        isRefineOnMapMove: true,
+      };
+
+      const wrapper = shallow(<Provider {...props} />);
+      const mapValue = wrapper.instance().mapValue;
```

Comment:
We heavily rely on implementation details here, would have been better to use a `Consumer` to test those cases. We also miss some cases e.g. when `isRefineOnMapMove` is equal. We can improve those tests later.